ci/ci/build.py

```diff
@@ -93,7 +94,7 @@ def request(step):
 
             for s in requested_steps:
                 request(s)
-            self.steps = [s for s in self.steps if s in visited]
+            self.steps = [s for s in self.steps if s.name in visited]
```

Comment:
Shouldn't this still be `s in visited`?  We're doing `visited.add(step)` not `visited.add(step.name)` above.

Reply:
Yup, changed this a different way at first and forgot to back this out.